### DISCUSSION TITLE
GNNExplainer using a batch of data output differently when using a single data

### DISCUSSION THREAD

**Discussion by JPaulYang** [pyg-team/pytorch_geometric#10099] [General]
Hi,

I've been working with the explainer class for a while now (`torch_geometric==2.6.0`) and I noticed the generated edge mask for using one pyg data is not quite the same as the one using, for example, a batch of two pyg data objects, after 200 epochs of training. 

The configuration was like this:
```
   explainer = Explainer(
        model=gnn,
        algorithm=GNNExplainer(epochs=num_epochs),
        explanation_type='model',
        edge_mask_type='object',
        model_config=dict(
            mode='multiclass_classification',
            task_level='graph',
            return_type='raw',
        ),
    )
```
Then I train the explainer for 200 epochs for example.
```
explanation_1 = explainer(data1.x, data1.edge_index, edge_attr=data1.edge_attr, batch=data1.batch)
explanation_2 = explainer(data2.x, data2.edge_index, edge_attr=data2.edge_attr, batch=data2.batch)

combined_data = Batch.from_data_list([data1, data2])
explanation_combined = explainer(combined_data.x, combined_data.edge_index, edge_attr=combined_data.edge_attr, batch=combined_data.batch)
```
The edge masks from one data instance is stable under different seeds, however, when combine it with another data, the edge mask on the same graph seems to deviate from the previous output. Could anyone provide any insights on that? Any helps will be much appreciated.
